0.5.quality_control/0.jump_umap_analysis_with_cosmicqc.py

```diff
@@ -0,0 +1,698 @@
+    for record in df_merged_single_cells[
+        [
+            "Metadata_Well",
+            "Image_Metadata_Row",
+            "Image_Metadata_Col",
+            "Image_Metadata_Site",
+        ]
+    ].to_dict(orient="records")
+}
+# fmt: on
+
+next(iter(outline_to_orig_mapping.items()))
+# -
+
+# label outliers within the dataset
+print("Large nuclei outliers:")
+df_labeled_outliers = cosmicqc.analyze.find_outliers(
+    df=df_merged_single_cells,
+    metadata_columns=metadata_cols,
+    feature_thresholds="large_nuclei",
+)
+
+# label outliers within the dataset
+print("Elongated nuclei outliers:")
+df_labeled_outliers = cosmicqc.analyze.find_outliers(
+    df=df_merged_single_cells,
+    metadata_columns=metadata_cols,
+    feature_thresholds="elongated_nuclei",
+)
+
+# label outliers within the dataset
+print("Small and low formfactor nuclei outliers:")
+df_labeled_outliers = cosmicqc.analyze.find_outliers(
+    df=df_merged_single_cells,
+    metadata_columns=metadata_cols,
+    feature_thresholds="small_and_low_formfactor_nuclei",
+)
+
+# label outliers within the dataset
+df_labeled_outliers = cosmicqc.analyze.label_outliers(
+    df=df_merged_single_cells,
+    include_threshold_scores=True,
+)
+# show added columns
+df_labeled_outliers[
+    [col for col in df_labeled_outliers.columns.tolist() if "cqc." in col]
+].head()
+
+# show small and low formfactor nuclei outliers
+CytoDataFrame(
+    data=pd.DataFrame(df_labeled_outliers).sort_values(by="cqc.small_and_low_formfactor_nuclei.is_outlier", ascending=False),
+    data_context_dir="../0.download_data/data/images/orig",
+    data_outline_context_dir="../0.download_data/data/images/outlines",
+    segmentation_file_regex=outline_to_orig_mapping,
+)[
+    [
+        "cqc.small_and_low_formfactor_nuclei.is_outlier",
+        "Image_FileName_OrigAGP",
+        "Image_FileName_OrigDNA",
+        "Image_FileName_OrigRNA",
+    ]
+]
+
+# +
+# create a column which indicates whether an erroneous outlier was detected
+# from all cosmicqc outlier threshold sets. For ex. True for is_outlier in
+# one threshold set out of three would show True for this column. False for
+# is_outlier in all threshold sets would show False for this column.
+df_labeled_outliers["analysis.included_at_least_one_outlier"] = df_labeled_outliers[
+    [col for col in df_labeled_outliers.columns.tolist() if ".is_outlier" in col]
+].any(axis=1)
+
+# show value counts for all outliers
+outliers_counts = df_labeled_outliers[
+    "analysis.included_at_least_one_outlier"
+].value_counts()
+outliers_counts
+# -
+
+# show the percentage of total dataset
+print(
+    (outliers_counts.iloc[1] / outliers_counts.iloc[0]) * 100,
+    "%",
+    "of",
+    outliers_counts.iloc[0],
+    "include erroneous outliers of some kind.",
+)
+
+# ## Prepare data for analysis with pycytominer
+
+# +
+parquet_sampled_with_outliers = f"./{example_plate}_sampled_with_outliers.parquet"
```

Comment:
Where does this come from? I might have missed something being saved above.

Reply:
Thanks for checking, this is a filename which is used to either read or create a file using the same name. We reference it here first as it will be used to create or read the file below this line.